rero_ils/modules/stats/tasks.py

```diff
@@ -26,11 +27,12 @@
 def collect_stats_billing():
     """Collect and store the statistics for billing."""
     stats_pricing = StatsForPricing()
-    stat = Stat.create(
-        dict(type='billing', values=stats_pricing.collect()),
-        dbcommit=True, reindex=True)
-    return f'New statistics of type {stat["type"]} has\
-        been created with a pid of: {stat.pid}'
+    with current_app.app_context():
+        stat = Stat.create(
+            dict(type='billing', values=stats_pricing.collect()),
```

Comment:
This will not solve the db timeout, you need to compute the collect outside of the app_context,